spec/rubocop/cop/root_cops/factory_file_name_spec.rb

```diff
@@ -37,19 +65,38 @@
       end
     end
 
-    context "when there are closely related factories in the same file" do
+    context "and the factory uses the file name as a prefix" do
       it "does not register an offense" do
         expect_no_offenses(<<-RUBY.strip_indent)
           FactoryBot.define do
-            factory :factory_name do
+            factory :user_billings__payment do
               ref { "new-branch" }
               ref_type { "branch" }
               master_branch { "master" }
             end
           end
+        RUBY
+      end
+    end
 
+    context "and the factory name uses an incorrect prefix" do
+      it "registers an offense" do
+        expect_offense(<<-RUBY.strip_indent)
           FactoryBot.define do
-            factory :factory_name__related_factory do
+            factory :user_billing__payment do
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Factory name uses incorrect prefix, should be 'user_billings__payment'.
+              ref { "new-branch" }
+              ref_type { "branch" }
+              master_branch { "master" }
+            end
+            factory :users_billing__payment do
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Factory name uses incorrect prefix, should be 'user_billings__payment'.
+              ref { "new-branch" }
+              ref_type { "branch" }
+              master_branch { "master" }
+            end
+            factory :users_billings__payment do
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Factory name uses incorrect prefix, should be 'user_billings__payment'.
```

Comment:
This is pretty picky, but maybe have these be different names?

Reply:
One is `billing` and the other is `billings`. I think I was just making sure that all incorrect combinations of singular/plural would throw and error.

Reply:
My bad, I totally missed that difference. All good here!